cmake/modules/extensions.cmake

```diff
@@ -3425,6 +3426,38 @@ function(target_sources_if_dt_node path target scope item)
   endif()
 endfunction()
 
+########################################################
+# 4.3 zephyr_dt_*
+#
+# The following methods are common code for dealing
+# with devicetree related files in CMake.
+#
+# Note that functions related to accessing the
+# *contents* of the devicetree belong in section 4.1.
+# This section is just for DT file processing at
+# configuration time.
+########################################################
+
+# Usage:
+#   zephyr_dt_normalize_overlay_list(FOO_OVERLAY_FILE FOO_OVERLAY_FILE_AS_LIST)
+#
+# Converts the argument to a ;-list with CMake path names, after
+# passing its contents through a configure_file() transformation.
+#
+# <var>     : Input variable name to normalize
+# <out_var> : Output variable that will be set to the final list.
+function(zephyr_dt_normalize_overlay_list var out_var)
```

Comment:
> Maybe call it: zephyr_list() similar to how we extend the CMake string() and file() in the zephyr_string() and zephyr_file() functions.

I'll wait for your responses to the above review comments since it looks like you need some more time to let this one soak. I'll push an update with a zephyr_list helper with any other necessary changes from those comments.